assets/getWords.js

```diff
@@ -0,0 +1,137 @@
+const statsText = document.getElementById('stats-text');
+const inputButton = document.getElementById('generate-text-button');
+var paragraphWithText = document.getElementById('text');
+
+function cleanText(text) {
+  const resultText = text
+    .replaceAll(',', " ")
+    .replaceAll('"', "")
+    .replaceAll('[', "")
+    .replaceAll(']', "");
+
+  return resultText;
+}
+
+function charArrayIntoString(textArray) {
+  var resultString = "";
+  for (const char of textArray) {
+    resultString += char;
+  }
+
+  return resultString;
+}
+
+function markCurrentChar(paragraphWithText, charIndex) {
+  let span = document.createElement('span');
+  let textForSpan = document.createTextNode(paragraphWithText.innerText[charIndex]);
+  span.appendChild(textForSpan);
+  span.style.backgroundColor = 'white';
+  let resultParagraphText = paragraphWithText.innerText.substring(0, charIndex) + span.outerHTML + paragraphWithText.innerText.substring(charIndex + 1);
+
+  return resultParagraphText;
+}
+
+function markIncorrectChar(paragraphWithText, charIndex) {
+  let span = document.createElement('span');
+  let textForSpan = document.createTextNode(paragraphWithText.innerText[charIndex]);
+  span.appendChild(textForSpan);
+  span.style.color = 'red';
+  let resultParagraphText = paragraphWithText.innerText.substring(0, charIndex) + span.outerHTML + paragraphWithText.innerText.substring(charIndex + 1);
+
+  return resultParagraphText;
+}
+
+var seconds = 0;
+var userMistakesCount = 0;
+var userKeyTypeCount = 0;
+var intervalIsUsed = false;
+
+displayStats(statsText, seconds, userMistakesCount, userKeyTypeCount);
+
+function incrementSeconds() {
+  intervalIsUsed = true;
+  seconds += 1;
+  displayStats(statsText, seconds, userMistakesCount, userKeyTypeCount);
+}
+
+generateTextButton.addEventListener('click', async function() {
+  seconds = 0;
+  displayStats(statsText, seconds, userMistakesCount, userKeyTypeCount);
+
+  if (!intervalIsUsed) {setInterval(incrementSeconds, 1000);}
+
+  let textArray = [];
+  const amountOfWordsWanted = document.getElementById('amount-of-words').value;
+  const amountOfWordsWantedToString = amountOfWordsWanted.toString();
+  const webAPI = 'https://random-word-api.herokuapp.com/word?number=' + amountOfWordsWantedToString;
+  const fetchText = await fetch(webAPI);
+  const text = await fetchText.text();
+  const resultText = cleanText(text);
+
+  for (const element of resultText) {
+    textArray.push(element);
+  }
+
+  paragraphWithText.innerHTML = charArrayIntoString(textArray);
+
+  var charIndex = 0;
+  paragraphWithText.innerHTML = markCurrentChar(paragraphWithText, charIndex);
+
+  inputButton.addEventListener('keydown', function(event) {
+    if (event.code == 'Space') {
+      event.preventDefault();
+
+      return;
+    }
+  });
+
+  resetProgressButton.addEventListener('keydown', function(event) {
+    if (event.code == 'Space') {
+      event.preventDefault();
+
+      return;
+    }
+  });
```

Comment:
Isn't this a double from above? Do you really need this?

Reply:
This isn't entirely a double, each of with are called for another button. I want to prevent if I click on one of these buttons, the space bar should do nothing. That is for the inputButton and the resetProgressButton.